File: Android/Sales/app/src/main/java/com/trimline/sales/Converters.java
package com.trimline.sales;

import androidx.room.TypeConverter;

import java.sql.Date;
import java.sql.Time;

public class Converters {
    public static class DateConverter {

        @TypeConverter
        public static Date toDate(Long dateLong){
            return dateLong == null ? null: new Date(dateLong);
        }

        @TypeConverter
        public static Long fromDate(Date date){
            return date == null ? null : date.getTime();
        }
    }
    public static class TimeConverter {

        @TypeConverter
        public static Time toTime(Long dateLong){
            return dateLong == null ? null: new Time(dateLong);
        }

        @TypeConverter
        public static Long fromTime(Time date){
            return date == null ? null : date.getTime();
        }
    }


}

File: Android/Sales2/app/src/main/java/com/trimline/sales/Converters.kt
package com.trimline.sales

import android.util.Log

import androidx.room.TypeConverter
import java.sql.Date
import java.sql.Time
import java.text.DateFormat
import java.text.SimpleDateFormat

class Converters {
    @TypeConverter
    fun toDate(dateLong: Long?): Date? {
        return if (dateLong == null) null else Date(dateLong)
    }

    @TypeConverter
    fun fromDate(date: Date?): Long? {
        return date?.time
    }
@TypeConverter
fun toitemtype(itemtypes: Item_Type?): Int?{

    return itemtypes?.ordinal;
}
    @TypeConverter
    fun fromitemtype(itemtypes: Int?): Item_Type?{
        return Item_Type.values()[itemtypes!!];
    }
}